addons/mail/models/mail_message.py

```diff
@@ -659,9 +659,16 @@ def unlink(self):
         self.mapped('attachment_ids').filtered(
             lambda attach: attach.res_model == self._name and (attach.res_id in self.ids or attach.res_id == 0)
         ).unlink()
+        msg_ids_by_partner = defaultdict(list)
         for elem in self:
             if elem.is_thread_message():
                 elem._invalidate_documents()
+            for partner in self.partner_ids:
```

Comment:
This list of partners is probably not what you would expect.

More often than not it will contain customers, for which we don't care to send the notification.
It also contains mentioned partners, which is ok (for inbox).

But it will not contain people that have starred, that received the message through followers mechanism (with inbox enabled), etc.

Reply:
Yes indeed, it is not self.partner_ids but elem.partner_ids. 
Also doing it just after the invalidation was not optimal so I moved it above.
I have also filtered the notification sent to the client to only send the notification to clients using inbox at the cost of a few added queries.